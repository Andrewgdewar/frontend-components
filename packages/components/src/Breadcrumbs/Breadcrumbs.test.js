import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Breadcrumbs from './Breadcrumbs';

describe('Breadcrumbs component', () => {
  const onNavigate = jest.fn();
  const items = [
    { title: 'First', navigate: 'first' },
    { title: <span>Second</span>, navigate: 'second' },
  ];
  const current = 'Something';
  describe('should render correctly', () => {
    it('no items', () => {
      const { container } = render(<Breadcrumbs items={undefined} current={current} onNavigate={onNavigate} />);
      expect(container).toMatchSnapshot();
    });

    it('no current', () => {
      const { container } = render(<Breadcrumbs items={items} current={undefined} onNavigate={onNavigate} />);
      expect(container).toMatchSnapshot();
    });

    it('with items and current item', () => {
      const { container } = render(<Breadcrumbs items={items} current={current} onNavigate={onNavigate} />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('API', () => {
    it('should call correctly', () => {
      const { container } = render(<Breadcrumbs items={items} current={current} onNavigate={onNavigate} />);
      userEvent.click(container.querySelector('a'));
      expect(onNavigate.mock.calls.length).toBe(1);
      expect(onNavigate.mock.calls[0][1]).toBe(items[0].navigate);
      expect(onNavigate.mock.calls[0][2]).toBe(0);
    });
  });
});
