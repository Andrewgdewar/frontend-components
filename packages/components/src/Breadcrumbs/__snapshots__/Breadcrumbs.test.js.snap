// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component should render correctly no current 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="pf-c-breadcrumb ins-c-breadcrumbs"
    data-ouia-component-id="OUIA-Generated-Breadcrumb-2"
    data-ouia-component-type="PF4/Breadcrumb"
    data-ouia-safe="true"
  >
    <ol
      class="pf-c-breadcrumb__list"
    >
      <li
        class="pf-c-breadcrumb__item"
        data-key="0"
      >
        <a
          aria-label="first"
        >
          First
        </a>
      </li>
      <li
        class="pf-c-breadcrumb__item"
        data-key="1"
      >
        <span
          class="pf-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
        <a
          aria-label="second"
        >
          <span>
            Second
          </span>
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Breadcrumbs component should render correctly no items 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="pf-c-breadcrumb ins-c-breadcrumbs"
    data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
    data-ouia-component-type="PF4/Breadcrumb"
    data-ouia-safe="true"
  >
    <ol
      class="pf-c-breadcrumb__list"
    >
      <li
        class="pf-c-breadcrumb__item"
      >
         
        Something
         
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Breadcrumbs component should render correctly with items and current item 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="pf-c-breadcrumb ins-c-breadcrumbs"
    data-ouia-component-id="OUIA-Generated-Breadcrumb-3"
    data-ouia-component-type="PF4/Breadcrumb"
    data-ouia-safe="true"
  >
    <ol
      class="pf-c-breadcrumb__list"
    >
      <li
        class="pf-c-breadcrumb__item"
        data-key="0"
      >
        <a
          aria-label="first"
        >
          First
        </a>
      </li>
      <li
        class="pf-c-breadcrumb__item"
        data-key="1"
      >
        <span
          class="pf-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
        <a
          aria-label="second"
        >
          <span>
            Second
          </span>
        </a>
      </li>
      <li
        class="pf-c-breadcrumb__item"
      >
        <span
          class="pf-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
         
        Something
         
      </li>
    </ol>
  </nav>
</div>
`;
